apache-maven/src/assembly/maven/bin/mvn.cmd

```diff
@@ -177,38 +177,25 @@ cd /d "%EXEC_DIR%"
 
 :endDetectBaseDir
 
+rem Initialize JVM_CONFIG_MAVEN_OPTS to empty to avoid inheriting from environment
+set JVM_CONFIG_MAVEN_OPTS=
+
 if not exist "%MAVEN_PROJECTBASEDIR%\.mvn\jvm.config" goto endReadJvmConfig
 
-@setlocal EnableExtensions EnableDelayedExpansion
-set JVM_CONFIG_MAVEN_OPTS=
-for /F "usebackq tokens=* delims=" %%a in ("%MAVEN_PROJECTBASEDIR%\.mvn\jvm.config") do (
-    set "line=%%a"
-
-    rem Skip empty lines and full-line comments
-    echo !line! | findstr /b /r /c:"[ ]*#" >nul
-    if errorlevel 1 (
-        rem Handle end-of-line comments by taking everything before #
-        for /f "tokens=1* delims=#" %%i in ("!line!") do set "line=%%i"
-
-        rem Trim leading/trailing spaces while preserving spaces in quotes
-        set "trimmed=!line!"
-        for /f "tokens=* delims= " %%i in ("!trimmed!") do set "trimmed=%%i"
-        for /l %%i in (1,1,100) do if "!trimmed:~-1!"==" " set "trimmed=!trimmed:~0,-1!"
-
-        rem Replace MAVEN_PROJECTBASEDIR placeholders
-        set "trimmed=!trimmed:${MAVEN_PROJECTBASEDIR}=%MAVEN_PROJECTBASEDIR%!"
-        set "trimmed=!trimmed:$MAVEN_PROJECTBASEDIR=%MAVEN_PROJECTBASEDIR%!"
-
-        if not "!trimmed!"=="" (
-            if "!JVM_CONFIG_MAVEN_OPTS!"=="" (
-                set "JVM_CONFIG_MAVEN_OPTS=!trimmed!"
-            ) else (
-                set "JVM_CONFIG_MAVEN_OPTS=!JVM_CONFIG_MAVEN_OPTS! !trimmed!"
-            )
-        )
-    )
-)
-@endlocal & set JVM_CONFIG_MAVEN_OPTS=%JVM_CONFIG_MAVEN_OPTS%
+rem Use Java to parse jvm.config to avoid batch script parsing issues with special characters
+rem This handles pipes, quotes, @, and other special characters correctly
+rem Use random temp directory to avoid conflicts between different Maven invocations
+set "JVM_CONFIG_PARSER_DIR=%TEMP%\mvn-jvm-parser-%RANDOM%-%RANDOM%"
+mkdir "%JVM_CONFIG_PARSER_DIR%"
+set "JVM_CONFIG_TEMP=%TEMP%\mvn-jvm-config-%RANDOM%.txt"
+"%JAVACMD:java.exe=javac.exe%" -d "%JVM_CONFIG_PARSER_DIR%" "%MAVEN_HOME%\bin\JvmConfigParser.java" >nul 2>&1
```

Comment:
Same as on posix/unix: why run `javac`? We are on 17 and can easily run single-source files using `java`.